### ISSUE TITLE
Sporadically appearing "request timed out" 

### ISSUE THREAD

**Issue by sirvaulterscoff** [coreos/jetcd#297]
We're running java 8 spring boot application on etcd3 and we're getting "StatusRuntimeException: INTERNAL: etcdserver: request timed out" from time to time. Both etcd3 and jetcd seems to have not time-out options. Thus totally unclear what's causing the issue.

We're running the same application on another box and it shows no signs of those errors.
Stacktrace follows
```
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: INTERNAL: etcdserver: request timed out
        at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:299)
        at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:286)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116)
        at com.coreos.jetcd.internal.impl.Util.lambda$toCompletableFutureWithRetry$1(Util.java:125)
        ... 3 common frames omitted
Caused by: io.grpc.StatusRuntimeException: INTERNAL: etcdserver: request timed out
        at io.grpc.Status.asRuntimeException(Status.java:526)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:467)
        at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
        at com.coreos.jetcd.internal.impl.ClientConnectionManager$AuthTokenInterceptor$1$1.onClose(ClientConnectionManager.java:280)
        at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
        at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1$1.onClose(CensusStatsModule.java:684)
        at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:41)
        at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1$1.onClose(CensusTracingModule.java:392)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:475)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:63)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:557)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:478)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:590)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        ... 3 common frames omitted
```

Any thoughts on what can cause those errors?

**Comment by fanminshi** [comment]
@sirvaulterscoff could you show you me the code that is causing the timeout?
If you can provide reproducible steps, that will help me debugging this more.

**Comment by sirvaulterscoff** [comment]
Sorry for opening the issue too soon, appered to be resolved by upgrading etcd to 3.3.2
